container/image_test.py

```diff
@@ -760,6 +760,189 @@ def test_war_image(self):
         './jetty/webapps/ROOT/WEB-INF/lib/javax.servlet-api-3.0.1.jar',
       ])
 
+  def test_nodejs_image(self):
+    self.maxDiff = None
+    with TestImage('nodejs_image') as img:
+      # TODO: remove all '/app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/external'
+      # once --noexternal_legacy_runfiles is enabled
+      # https://github.com/bazelbuild/rules_docker/issues/1350
+
+      # Check the application layer (top layer), which also contains symlinks to the bottom layers.
+      self.assertTopLayerContains(img, [
+        '.', 
+        './app', 
+        './app/testdata', 
+        './app/testdata/nodejs_image_binary.runfiles', 
+        './app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker', 
+        './app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/testdata', 
+        './app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/testdata/nodejs_image.js', 
+        './app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/testdata/nodejs_image_lib.js', 
+        './app/testdata/nodejs_image_binary.runfiles/io_bazel_rules_docker/testdata/nodejs_image_lib.d.ts', 
+        './app/testdata/nodejs_image_binary.runfiles/build_bazel_rules_nodejs', 
+        './app/testdata/nodejs_image_binary.runfiles/build_bazel_rules_nodejs/internal', 
+        './app/testdata/nodejs_image_binary.runfiles/build_bazel_rules_nodejs/internal/linker', 
```

Comment:
I'd like to put the files here from rules_nodejs into their own layer. Making this top layer only actual source files / generates files under the `io_bazel_rules_docker` workspace.  
But even with how it looks right now is a big improvement

Reply:
Will adress this one another time